src/Symfony/Component/Security/Http/Firewall.php

```diff
@@ -53,13 +54,21 @@ public function onKernelRequest(GetResponseEvent $event)
             $exceptionListener->register($this->dispatcher);
         }
 
+        $hasResponse = false;
```

Comment:
You don't need to create this variable. Just use `$event->hasResponse()` in the right place.